### ISSUE TITLE
Every deposited document is assigned a persistent identifier (ARK) [5]

### ISSUE THREAD

**Issue by sebastiendeleze** [rero/sonar#399]
When a document is created, it will be assigned a persistent identifier (ARK).

**Data model**
The ARK identifier must be stored in the `identifiedBy` property, like the following example:

```json
{
    "identifiedBy": [
        {
            "value": "ark:/99999/ffk37p4d",
            "type": "bf:Local",
            "source": "ARK"
        }
    ]
}
```

A flag is set to know that the identifier has been created automatically and thus cannot be modified in the editor. This flag can be the creation date or a boolean.

**Editor**

The ARK identifier cannot be edited.

**Search**

The ARK identifier is searchable in the identifier index.

**Display**
In the document detailed view (public + admin), The identifier is displayed like this ("Permalink" renamed in "Persistent URL"):
![image](https://user-images.githubusercontent.com/8154915/114506455-1801c880-9c32-11eb-9eaf-6b9b83a19b20.png)

The persistent URL is a link that displays the current document and the ARK identifier is kept in the URL (no redirection to `/global/documents/XXX`).

Previous proposal:
![ark](https://user-images.githubusercontent.com/6112685/114410088-5d2ee780-9bab-11eb-97ae-798773c2d4d4.png)

**API**

The API documentation is available here: https://www.arketype.ch/doc/api.
Here's the info to test the API:
Account: **apitest**
Shoulder: **99999/ffk3**
Password: *communicated separately*
This is an access only for testing purpose.

**Comment by sebastiendeleze** [comment]
@pronguen Questions concerning the URL. 
1. The URL with ARK identifier should be redirected to the `standard` URL (https://sonar.ch/ark:/99999/ffk37p4d --> https://sonar.ch/global/documents/xxx)?
2. Should this URL be displayed in the `Permalink` section or, as specified in mockup, in `Persitent URL` section?

**Comment by pronguen** [comment]
@sebastiendeleze 

>     1. The URL with ARK identifier should be redirected to the `standard` URL (https://sonar.ch/ark:/99999/ffk37p4d --> https://sonar.ch/global/documents/xxx)?

I would say yes. Any other options?

>     2. Should this URL be displayed in the `Permalink` section or, as specified in mockup, in `Persitent URL` section?

In "Persistent URL" if possible.

For the context, we are going to abandon all RERO Explore permalinks. With ARK in SONAR, this situation should be avoided, and identifiers will be really persistent even if we change to a new system. So: permalink is not a real promise, but persistent URL/ID it is.

**Comment by sebastiendeleze** [comment]
> For the context, we are going to abandon all RERO Explore permalinks. With ARK in SONAR, this situation should be avoided, and identifiers will be really persistent even if we change to a new system. So: permalink is not a real promise, but persistent URL/ID it is.

> > ```
> > 1. The URL with ARK identifier should be redirected to the `standard` URL (https://sonar.ch/ark:/99999/ffk37p4d --> https://sonar.ch/global/documents/xxx)?
> > ```
> 
> I would say yes. Any other options?
> 

We can keep the URL as is (https://sonar.ch/ark:/99999/ffk37p4d) and display directly the document detail without a redirection.

**Comment by jma** [comment]
I have several questions/propositions:

- the ark link should be: https://n2t.net and not sonar.ch
- we should not generate an ark for harvested  records as we have the deduplication (deleting an ark is not possible)
- do we include the view code in the resolved URL? 'global' for shared?
  - if yes we need to change the resolved URL when the institution is changed
- how to deal with `hiddenFromPublic`, is still used?
- do we put also the metadata? => need to update the metadata at each time the record is changed
- what to do when a record is removed?

**Comment by mmo** [comment]
> I have several questions/propositions:
> 
> * the ark link should be: https://n2t.net and not sonar.ch

The [ARK Identifiers FAQ](https://wiki.lyrasis.org/display/ARKs/ARK+Identifiers+FAQ) provides useful information which helps explain this choice, namely the following two items.

(emphasis added)

[How do I cite or advertise an ARK?](https://wiki.lyrasis.org/display/ARKs/ARK+Identifiers+FAQ#ARKIdentifiersFAQ-HowdoIciteoradvertiseanARK?)

> The URL (https or http) form of the ARK is preferred, for example,
>         https://n2t.net/ark:/99166/w66d60p2
> An ARK meant for external use is generally advertised (released, published, disseminated) in this way in order to be an actionable identifier.
> (...)
> **An important decision is whether your URL-based ARKs will use the hostname of your local resolver or the N2T.net resolver.** If local control or branding is important enough, you would advertise ARKs based at your local resolver. If you're concerned about the stability of your local hostname, you would advertise your ARKs based at n2t.net.
> 
> Resolving your ARKs through N2T is always possible for users, regardless of how you advertise them.


[If most ARKs run on their own resolvers, why is there also a global resolver for ARKs?](https://wiki.lyrasis.org/display/ARKs/ARK+Identifiers+FAQ#ARKIdentifiersFAQ-IfmostARKsrunontheirownresolvers,whyistherealsoaglobalresolverforARKs?)

> Most ARKs are created by organizations that advertise ("publish") them based at their own resolvers. For example, this ARK was published based at the ark.bnf.fr resolver:
>           http://ark.bnf.fr/ark:/12148/btv1b8449691v/f29
> **Having to run and maintain your own resolver is the cost of complete autonomy.** Using your own resolver also lets you do branding via the hostname, **the downside being that brands are transient and tend to make identifiers fragile**. Political and even legal (eg, trademarks) pressures may make supporting older branded hostnames, hence their identifiers, difficult.

**Comment by mmo** [comment]
> I have several questions/propositions:
> 
> * **what to do when a record is removed?**

The identifier's [`_status`](https://ezid.cdlib.org/doc/apidoc.html#internal-metadata) field should be set to `unavailable` and its metadata be removed. See the [section "Identifier status" in the EZID API documentation](https://ezid.cdlib.org/doc/apidoc.html#identifier-status).

Ideally, we should provide a reason for the unavailability, such as "resource deleted". The API seems to support this: instead of setting the status to `unavailable`, we set it to `unavailable | resource deleted`. Cf. [section "Identifier status"](https://ezid.cdlib.org/doc/apidoc.html#identifier-status): *A reason for the object's unavailability may optionally follow the status separated by a pipe character ("|", U+007C), e.g., "unavailable | withdrawn by author"*.

**Comment by jma** [comment]
Sample of ark request:
```python
import requests
from requests.auth import HTTPBasicAuth

res = requests.post('https://www.arketype.ch/shoulder/ark:/99999/ffk3', auth=HTTPBasicAuth('user', 'password'), data='
    ...: _target: http://www.sonar.ch/global/records/1')
```
